doc/development/contributing.rst

```diff
@@ -85,6 +85,15 @@ Cython will detect and compile only the files that have been changed, so this co
    When undertaking Cython development, the reason we use ``python setup.py develop`` instead of ``pip install -e .`` is because Cython's changed-file detection does not reliably work in the latter.
    ``pip`` tends to build in temporary virtual environments, which often makes Cython think its core library files have been updated, triggering a complete, slow rebuild of everything.
 
+.. note::
+
+   When developing for a new major or minor version (not a patch release), new feature introduced could require newer versions of some of the required packages.
+   In that case, update the requirements as needed.
+   We follow `NEP29`_ for minimum supported versions of scipy and numpy.
+   For other dependencies, we support the oldest version that is supporting the suggested python and numpy's versions.
+
+.. _NEP29: https://numpy.org/neps/nep-0029-deprecation_policy.html
+
```

Comment:
I think this information is mostly necessary only for core developers. I would suggest to keep only the documentation on `reslease_distribution.rst`. This seems to also be the approach followed by scipy for instance.

Reply:
I was not sure where to add it, so I added it in both. But I think it could be useful to have something here:
If we say that we support python >=3.8, that mean we can now use the `:=` operator, but not `dict | dict` which was introduced in python 3.9.
This information should be available to contributors.

Reply:
I see your point. However, I think that the average contributor will care only about the specific supported versions, not the procedure employed to select them. The supported versions will be ultimately dictated by tests, so if tests fail we can discuss in the appropriate PR to update the  tests dependencies and run-time requirements. What do you think about using instead the following note:
```suggestion
.. note::

QuTiP follows `NEP29`_ when selecting the supported version of its dependencies.
The currently supported versions can be checked in the `setup.cfg`.
These coincide with the versions employed for testing in continuous integration.

In the event of a feature requiring a version upgrade of python or a dependency, it will 
be considered appropriately in the pull request.
In any case, python and dependency upgrades will only happen in mayor or minor 
versions of QuTiP, not in a patch.

.. _NEP29: https://numpy.org/neps/nep-0029-deprecation_policy.html
```
Feel free to reject this suggestion if you think is not necessary.

Reply:
`setup.cfg` should list what we actually support, not what we want to support. 

But I agree that having a specific version written somewhere would be useful. I will put it in the road-map, this show that it is more a target than something fixed: if we released a new version last week, we would have to support numpy 1.19.